### ISSUE TITLE
Test idling on a hosted Che

### ISSUE THREAD

**Issue by Katka92** [eclipse-che/che#16511]
### Is your task related to a problem? Please describe.
Test that workspace is idling correctly on a hosted Che.
This test should be part of pre-release testing, not a periodic test suite.

### Describe the solution you'd like
Create a workspace and check that after the exact time period, the workspace is switched to offline.
There are two possible solutions:
1. Create a workspace and check its state via API calls
2. Create and check workspace status via UI (user should see the yellow status bar informing about the offline status of a workspace).

**Comment by Katka92** [comment]
I was thinking about that for a while and I think I like to test it both ways. It means use typescript tests to create a workspace via UI, then check the status via API calls. Once the workspace is STOPPED, check the UI, if there is a yellow bottom bar that indicates that the workspace is stopped. 
As @ibuziuk said, it is true that having that as a part of some suite will highly increase time for that suite. So we have two options then - we can run that in parallel or have a separate suite for that. I would go for having a special job that will include just the idling test. It will run nightly. We can also create a mechanism that will allow us to run that test via PR comment and specify which deployment we want to test - so we can use that same way as pre-release test suite. 
I would like to hear more opinions on that, so feel free to comment guys :)

**Comment by ibuziuk** [comment]
@Katka92 @rhopp I believe as the first it might be a good idea to create a test for upstream and run it as part of nightly suite and after that reuse as-is for Hosted Che

**Comment by Katka92** [comment]
@ibuziuk Yes, sure the implementation will be done in upstream. Sorry for not specifying that. It also can be added to the nightly upstream suite. But on the Hosted Che side we can still decide how to use that test - e.g. have a separate job for that.

**Comment by ibuziuk** [comment]
@Katka92 for Hosted Che I would probably create a separate nightly job with all the e2e tests we have and also include the idling test there.

**Comment by Katka92** [comment]
I'm not for having a nightly job with all e2e tests. Even now we spend ~1 hour per day going through the results of periodic tests. Having nightly job that will include all e2e tests will mean extra (and not small) time amount. That's why I would go just for having idling as a nightly and having all e2e tests in the pre-release test suite. Would be nice to have some comments from @ScrewTSW and @rhopp too.

**Comment by ibuziuk** [comment]
>  Even now we spend ~1 hour per day going through the results of periodic tests. Having nightly job that will include all e2e tests will mean extra (and not small) time amount

That's a lot of effort in my opinion. Can we instead prioritize the stability of tests in the first place? I would expect no false-positive results and failures only on the real issues

**Comment by ScrewTSW** [comment]
Do we really need to verify the idling nightly if there are no changes in the deployment? I mean if the idling passed during release, it should still work, right?
If above is true, I would just run it as part of the pre-release test suite and execute it in parallel with a separate account so that we can have the idling test running while we verify other devfiles and then check for result at the end of the suite.

**Comment by Katka92** [comment]
@ibuziuk do you see any value in testing idling specifically on each cluster (meaning 4 prod + 1 prod-previw)? If we would go for all clusters - I think having a separate nightly (or weekly) job with idling would be the way to go. 
For testing idling during pre-release testing I think the best way would be to create separate job and trigger that by separate command the same way as pre-release tests:
```
<url_to_be_tested>
[test-idling]
```
That way we would be able to run pre-release tests and idling tests at one time easily. Then if we will need to retest e.g. idling, we don't have to run devfile tests and vice versa.

**Comment by ibuziuk** [comment]
> do you see any value in testing idling specifically on each cluster (meaning 4 prod + 1 prod-previw)

no

> For testing idling during pre-release testing I think the best way would be to create separate job and trigger that by separate command the same way as pre-release tests:

ideally I would like to see the idling part of the pre-release test suite

**Comment by Katka92** [comment]
After a discussion with @flacatus and @rhopp it was decided that the test will be part of the test harness suite [1]. This test will be API based (not checking UI) and will be able to run against Che too. New job will be created for Hosted Che but the trigger would be the same as for the typescript pre-release tests. From the developer's point of view, all would remain the same - dev will comment just `<URL> pre-release-tests` and the test suite will be executed in parallel. 
The plan is to start with implementation next sprint.

[1] https://github.com/che-incubator/che-test-harness

**Comment by Katka92** [comment]
Idling test was [added](https://github.com/che-incubator/che-test-harness/pull/9) to upstream (the next step is to create [automation](https://github.com/eclipse/che/issues/18029)). We're not planning to run that test against Hosted Che currently - closing this issue.
